### PR TITLE
Programming Guide

### PR DESCRIPTION
**Pull request by liufengyun**
This is a work in progress.

### CODE REVIEW

File: docs/code/simple.scala [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+val rules = new MarketRules()
+
+// Create components
+val market = builder.createRef(Props(classOf[OrderBookMarketSimulator], marketId, rules), "market")
+val sobiTrader = builder.createRef(Props(classOf[SobiTrader], 123L, 3000, 2, 700.0, 50, 100.0, rules), "sobiTrader")
+
+// Connect components
+sobiTrader -> (market, classOf[LimitBidOrder], classOf[LimitAskOrder])
+
+builder.start
```
Comment (merlinND) [inline docs/code/simple.scala:18]: It's really nice to have included the code sample from a source file! It's clean.

**Comment by merlinND** [comment]
:sparkles:

File: docs/content/simulation.tex [hunk 2]
```diff
@@ -0,0 +1,164 @@
+\section{Simulation}
+
```
Comment (merlinND) [inline docs/content/simulation.tex:9]: As an introduction, explain quickly what we are trying to achieve with this "simulation" thing

File: docs/content/simulation.tex [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+\subsection{Simplifying the Real Market}
+Our project implements essential parts of the foreign exchange market.
+In order to set up a basic market simulator we make the following
+simplifications:
+
+\begin{itemize}
+    \item traders can only trade one symbol
```
Comment (merlinND) [inline docs/content/simulation.tex]: Define "instruments" and "spot transactions"

File: docs/content/simulation.tex [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+
+Since we are not able to answer these questions reliably and also since we cannot
+instantiate an infinite number of traders we introduce a different concept to
+initialize a simulation.
+
+We assign money to traders and market makers in a certain currency (e.g. USD) and 
+they receive the equivalent amount in the currency they need to trade their respective
+symbol.
+
+\begin{algorithm}
```
Comment (merlinND) [inline docs/content/simulation.tex:54]: _Nice_!

File: docs/content/simulation.tex [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+\begin{itemize}
+    \item the \textit{bid} price is well defined as the highest bid offer but the ask price is not
+    \item the market maker defines the \textit{ask} price by applying its own spread to the highest bid price
+    \item buying at market price means buying from the market maker at the highest \textit{bid price PLUS the spread}
+\end{itemize}
+
+If there are no limit orders at all:
+\begin{itemize}
+    \item ask and bid price stay constant
+    \item random traders hopefully prevent from freezing
```
Comment (merlinND) [inline docs/content/simulation.tex]: > hopefully prevent the simulation from freezing

**Comment by merlinND** [comment]
@mogli91 When you get the chance, could you add your sources as `\cite` or `\footnote` along the Simulation section?

File: docs/content/catalog.tex [hunk 6]
```diff
@@ -0,0 +1,312 @@
...
+\end{lstlisting}
+
+The fetcher reads the data from the local file system directly. It looks for directories named after the currency pair that's being fetched in the directory you supply to it. In the example above the source directory is \lstinline|data| and should be structured something like the following:
+
+\begin{lstlisting}
+data/
+*- EURCHF/
+|   ...
+|   *- DAT_NT_EURCHF_T_ASK_201304.csv
+|   *- DAT_NT_EURCHF_T_ASK_201304.txt
```
Comment (merlinND) [inline docs/content/catalog.tex]: Since we actually never use the `.txt`, I think we should remove them.

File: docs/content/catalog.tex [hunk 7]
```diff
@@ -0,0 +1,312 @@
...
+Note that all the fetchers in the table above are not components themselves, except \emph{HistDataCSVFetcher}. Non-component fetchers are used together with \emph{PullFetchComponent} as following snippet shows:
+
+\begin{lstlisting}[language=Scala]
+  val fetcherFx: TrueFxFetcher = new TrueFxFetcher
+  val fetcher = builder.createRef(Props(classOf[PullFetchComponent[Quote]], fetcherFx, implicitly[ClassTag[Quote]]), "TrueFxFetcher")
+\end{lstlisting}
+
+\subsubsection{HistDataCSVFetcher}
+
+This uses historical quote data from the free financial data provider histdata.com\footnote{More specifically http://www.histdata.com/download-free-forex-data/?/ninjatrader/tick-\{ask, bid\}-quotes}. The data we used can be comfortably fetched by using our get-data.sh script, for example like this: \lstinline|scripts/get-data.sh www.histdata.com EURCHF| (gets a set of CSV files to  \lstinline|./data/EURCHF|). HistDataCSVFetcher can take this data and send quotes "replaying" the past. To build a fetcher replaying EUR/CHF data from April 2013 through May 2013 with a speed factor of 60x (i.e. 1 hour is replayed in 1 minute) the following can be done:
```
Comment (merlinND) [inline docs/content/catalog.tex]: `get-data.sh`: make it use monospace font (I usually use `\texttt{}` but I see `\lstinline` seems to work well also)

**Comment by merlinND** [comment]
:shipit:

**Comment by liufengyun** [comment]
@merlinND thanks for merging it. If there're still missing parts or places to improve, I can do them tomorrow and before the deadline.

**Pull request merged**